src/Hosting/TestHost/src/HttpContextBuilder.cs

```diff
@@ -56,14 +61,24 @@ internal HttpContextBuilder(ApplicationWrapper application, bool allowSynchronou
 
         public bool AllowSynchronousIO { get; set; }
 
-        internal void Configure(Action<HttpContext> configureContext)
+        internal void Configure(Action<HttpContext, PipeReader> configureContext)
         {
             if (configureContext == null)
             {
                 throw new ArgumentNullException(nameof(configureContext));
             }
 
-            configureContext(_httpContext);
+            configureContext(_httpContext, _requestPipe.Reader);
+        }
+
+        internal void SendRequestStream(Func<PipeWriter, Task> sendRequestStream)
```

Comment:
I did it this way because the alternative is callbacks everywhere:
1. After request is complete
2. Request is aborted from server
3. Request is aborted from client

Sure HttpContextBuilder owns the request pipe and task, but using it is optional.

Reply:
Yea I don’t think this layering is super important